valohai_yaml/objs/endpoint.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import unicode_literals
+
+from .file import File
+
+
+class Endpoint(object):
+
+    def __init__(self, name, image, description=None, files=(), port=None, server_command=None, wsgi=None):
+        self.name = name
+        self.description = description
+        self.image = image
+        self.port = port
+        self.server_command = server_command
+        self.wsgi = wsgi
+
+        assert all(isinstance(f, File) for f in files)
+        self.files = files
+
+    @classmethod
+    def parse(cls, data):
+        kwargs = data.copy()
+        kwargs['files'] = [File.parse(f_data) for f_data in kwargs.pop('files', ())]
+        if 'server-command' in kwargs:
+            kwargs['server_command'] = kwargs.pop('server-command')
```

Comment:
Is there a better way to do this, i.e. parse the dash as an underscore?

Reply:
`_SimpleObject`'s `parse` and `serialize` methods do this roundtripping.

Reply:
So yeah, please just inherit from `_SimpleObject` and override parts of the `get_data()` method (mostly `files`).